include/SugarObjects/templates/basic/Basic.php

```diff
@@ -113,17 +159,16 @@ public function getEmailAddressConfirmOptIn($emailField)
 
         global $sugar_config;
 
-        if (!$sugar_config['email_enable_confirm_opt_in']) {
+        if (!$sugar_config['email_enable_confirm_opt_in']) {#
             global $log;
             $log->warn('Confirm Opt In is not enabled.');
-            return true;
+            return false;
         }
 
         $emailAddressId = $this->getEmailAddressId($emailField);
-        $emailAddress = BeanFactory::getBean('EmailAddresses', $emailAddressId);
-        $confirmOptIn = $emailAddress->confirm_opt_in;
-
-        return $confirmOptIn;
+        /** @var EmailAddress $emailAddressBean */
+        $emailAddressBean = BeanFactory::getBean('EmailAddresses');
+        return  $emailAddressBean->retrieve($emailAddressId);
```

Comment:
Illogical return value: The function called `getEmailAddressConfirmOptIn` return an `EmailAddress` object instead the status of Opt-in confirmation as it was initially. Rename the function to `getEmailAddress` or revert the return value to a `boolean`.